core/state/statedb_hooked.go

```diff
@@ -269,3 +269,18 @@ func (s *hookedStateDB) Finalise(deleteEmptyObjects bool) {
 		}
 	}
 }
+
+func (s *hookedStateDB) Hooks() *tracing.Hooks {
+	return s.hooks
+}
+
+// SetHooks provides a way to dynamically modify the Hooks,
+// since Hooks must be disabled during calls to `CreditFees` and `DebitFees`
+// Ref: contracts/fee_currencies.go
+func (s *hookedStateDB) SetHooks(hooks *tracing.Hooks) {
+	if hooks != nil {
+		s.hooks = hooks
+	} else {
+		s.hooks = new(tracing.Hooks)
+	}
+}
```

Comment:
To avoid merge conflicts when rebasing on a more recent upstream, we often move such additions to a file with a `celo_` prefix in the same go package. In this case that would be `core/state/celo_statedb_hooked.go`. This feels a bit cluttered for such small additions, but I still consider it an overall improvement. But I might be biased since I did a lot of rebasing this year.